dotnet/src/Functions/Functions.OpenApi/RestApiOperationRunner.cs

```diff
@@ -190,7 +190,14 @@ private static async Task<RestApiOperationResponse> SerializeResponseContentAsyn
     {
         var contentType = content.Headers.ContentType;
 
-        var mediaType = contentType?.MediaType ?? throw new KernelException("No media type available.");
+        // Check if content type is null for 204 response
+        if (contentType == null)
+        {
+            // Return an empty response with no content type for 204 responses
+            return new RestApiOperationResponse(string.Empty, string.Empty);
```

Comment:
Consider making both the Content and ContentType properties of the RestApiOperationResponse class nullable, and pass null instead of an empty string to indicate the absence of content or value.